server/src/main/java/org/elasticsearch/common/time/DateUtils.java

```diff
@@ -391,13 +390,17 @@ public static ZonedDateTime nowWithMillisResolution(Clock clock) {
 
     // check for all textual fields, and localized zone offset
     private static final Predicate<String> CONTAINS_CHANGING_TEXT_SPECIFIERS = System.getProperty("java.locale.providers", "")
-        .contains("COMPAT") ? Pattern.compile("[EcGaO]|MMM|LLL|eee|ccc|QQQ|ZZZZ").asPredicate() : Predicates.never();
+        .contains("COMPAT") ? Pattern.compile("[BEGOavz]|LLL|MMM|QQQ|ccc|eee|Z{4}").asPredicate() : Predicates.never();
```

Comment:
What is `B`? Should `v` be upper case (time-zone ID)?

Reply:
Can we have tests with these formats in DateUtilsTests to ensure we are checking/interpreting them correctly?

Reply:
`B` is 'Period of day' ('in the morning'). `v` is generic time-zone name (`V` is time-zone id), which I've seen can be localised